crates/bevy_asset/src/ser.rs

```diff
@@ -0,0 +1,87 @@
+use std::cell::Cell;
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{Asset, AssetServer, Handle};
+
+///////////////////////////////////////////////////////////////////////////////
+
+thread_local! {
+    static ASSET_SERVER: Cell<Option<AssetServer>> = Cell::new(None);
```

Comment:
So, IIRC, the handle path (from `get_handle_path` is stored in the `AssetServet` that thd handle was created from?
If so, this thread local asset server won't actually do anything, since all assets are created in the "global" `AssetServer` that you access in system via `Res<AssetServer>`

Reply:
there isn't global `AssetServer`, the `ASSET_SERVER` is used to pass the current asset server reference to each handle serializing functions. So if you call `serialize` on a asset instead of `asset_server.serialize_with_asset_refs` no handle will be serialized

Reply:
Yep, sry I was reading this on mobile and didn't see the `impl AssetServer` below.

Reply:
That would just not work, I'm pretty sure distill does a similar impl

Reply:
Yep, distill does a similar thing, however, I'm still not the biggest fan. 

If we could at least avoid the extra ref count (from `Arc`) via just using a `&` directly that would be much preferred.